File: dart/8.dart
import 'dart:io';

void main() {
  // 8
  String pocketMoneyPerMonth = stdin.readLineSync()!;
  String spentMoney = stdin.readLineSync()!;
  // work out how much is left
  String leftMoney = "${double.parse(pocketMoneyPerMonth)-double.parse(spentMoney)}"; // double is correct to use in this case 
  print("Spent $spentMoney and have $leftMoney left");
  // need to ask again as this is the task
  pocketMoneyPerMonth = stdin.readLineSync()!; // type has been set before
  // work out max spend per day
  String maxSpend = "${double.parse(pocketMoneyPerMonth)/30}";
  // output
  print("You can spend $maxSpend per day in a 30 day month!");
}

File: js/8.js
// 8th one i think
const prompt = require("prompt-sync")({ sigint: true });

let pocketMoneyPerMonth = prompt();
spentMoney = prompt();
// work out how much is left and output
let leftMoney = parseFloat(pocketMoneyPerMonth) - parseFloat(spentMoney);
console.log(`You have spent ${spentMoney} and have ${leftMoney} left`);
// ask the user again as that it what it asks for me to do for some reason 
pocketMoneyPerMonth = prompt();
// work out max spend per day
let maxSpendPerDay = parseFloat(pocketMoneyPerMonth)/30;
console.log(`You can spend ${maxSpendPerDay} per day`);